system-test/storage.js

```diff
@@ -1778,6 +1778,290 @@ describe('storage', function() {
       });
     });
 
+    describe('kms keys', function() {
+      var FILE_CONTENTS = 'secret data';
+
+      var keyRingId = generateName();
+      var cryptoKeyId = generateName();
+      var kmsKeyName;
+
+      var keyRingsBaseUrl = `https://cloudkms.googleapis.com/v1/projects/${
+        storage.projectId
+      }/locations/global/keyRings`;
+
+      function generateKmsKeyName(cryptoKeyId) {
+        return `projects/${
+          storage.projectId
+        }/locations/global/keyRings/${keyRingId}/cryptoKeys/${cryptoKeyId}`;
+      }
+
+      function createCryptoKey(cryptoKeyId, callback) {
+        storage.request(
+          {
+            method: 'POST',
+            uri: `${keyRingsBaseUrl}/${keyRingId}/cryptoKeys`,
+            qs: {cryptoKeyId},
+            json: {purpose: 'ENCRYPT_DECRYPT'},
+          },
+          function(err) {
+            if (err) {
+              callback(err);
+              return;
+            }
+
+            storage.request(
+              {
+                method: 'POST',
+                uri: `${keyRingsBaseUrl}/${keyRingId}/cryptoKeys/${cryptoKeyId}:setIamPolicy`,
+                json: {
+                  policy: {
+                    bindings: [
+                      {
+                        role: 'roles/cloudkms.cryptoKeyEncrypterDecrypter',
+                        members: `serviceAccount:${
```

Comment:
I recommend to use instead of writing the form `project-id@gs-project-accounts.iam.gserviceaccount.com` as this will change to `service-project-number@gs-project-accounts.iam.gserviceaccount.com`.
https://cloud.google.com/storage/docs/json_api/v1/projects/serviceAccount/get

Reply:
I'm not sure how to get the project number-- would the user running our tests have to explicitly give it to us, or is there an easier way? Auth for the tests comes from a service account JSON key file.

The API endpoint you gave only returns an email address, which uses the project ID:

```json
{
 "email_address": "nth-circlet-705@gs-project-accounts.iam.gserviceaccount.com",
 "kind": "storage#serviceAccount"
}
```

Reply:
The robot service account will be updated for GCS later this quarter and will be available using the serviceAccount.get request. My recommendation is to use the serviceAccount.get request to get the robot service account which will provide the correct format. You can ignore the new format for now.

Reply:
Done! https://github.com/googleapis/nodejs-storage/pull/179/commits/a1dce4f5d6041338b26862ee0aaba0be2e4ce3a7